File: src/main/java/de/tectoast/emolga/utils/records/StatLocation.java
package de.tectoast.emolga.utils.records;

import static de.tectoast.emolga.commands.Command.getAsXCoord;

@SuppressWarnings("unused")
public record StatLocation(String sheet, String x, int y) {
    private static final StatLocation INVALD = new StatLocation(null, 0, 0);

    public StatLocation(String sheet, int x, int y) {
        this(sheet, getAsXCoord(x), y);
    }

    public static StatLocation invalid() {
        return INVALD;
    }

    public boolean isValid() {
        return sheet != null;
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/records/StatLocation.kt
package de.tectoast.emolga.utils.records

import de.tectoast.emolga.commands.Command.Companion.getAsXCoord

//String sheet, String x, int y
class StatLocation(private val sheet: String?, val x: String, private val y: Int) {
    constructor(sheet: String?, x: Int, y: Int) : this(sheet, getAsXCoord(x), y)

    val isValid: Boolean
        get() = sheet != null

    override fun toString(): String {
        return "$sheet!$x$y"
    }

    companion object {
        private val INVALD = StatLocation(null, 0, 0)
        fun invalid(): StatLocation {
            return INVALD
        }
    }
}